subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -75,7 +75,7 @@ public JavaCompile() {
      * {@inheritDoc}
      */
     @Override
-    @PathSensitive(PathSensitivity.NAME_ONLY)
+    @PathSensitive(PathSensitivity.RELATIVE)
```

Comment:
I was surprised that only one test failed. I thought we still had some relocation tests which moved some classes around in the source directory.